drivers/gles2/shaders/scene.glsl

```diff
@@ -1562,9 +1562,13 @@ float sample_shadow(highp sampler2D shadow, highp vec4 spos) {
 
 #endif
 
-#if !defined(SHADOW_MODE_PCF_5) || !defined(SHADOW_MODE_PCF_13)
+#if !defined(SHADOW_MODE_PCF_5) && !defined(SHADOW_MODE_PCF_13)
+	
+	spos.xyz /= spos.w;
+	vec2 pos = spos.xy;
+	float depth = spos.z;
```

Comment:
Couldn't this be moved outside of the `#if defined(...)` blocks, since all 3 shadow filter modes now feature the same 3 lines?

Reply:
Yes go ahead please! I just copy/paste these code to minimal effecting to other codes

Reply:
Please make this change in this pull request so it can be merged :slightly_smiling_face:

Reply:
> Please make this change in this pull request so it can be merged 🙂

OK, compiling right now, will commit changes after I re-test https://github.com/godotengine/godot/issues/69759

Reply:
> please make a test build so I can re-test https://github.com/godotengine/godot/issues/54343 on the same android device

GitHub Actions already built one: https://github.com/godotengine/godot/suites/9779778662/artifacts/470173170

Reply:
> > please make a test build so I can re-test #54343 on the same android device
> 
> GitHub Actions has already built one: https://nightly.link/godotengine/godot/actions/artifacts/470173170.zip
> 
> See [Testing pull requests](https://docs.godotengine.org/en/latest/community/contributing/testing_pull_requests.html) for more information.

Just confirmed this fixed https://github.com/godotengine/godot/issues/54343

> > please make a test build so I can re-test #54343 on the same android device
> 
> GitHub Actions has already built one: https://nightly.link/godotengine/godot/actions/artifacts/470173170.zip
> 
> See [Testing pull requests](https://docs.godotengine.org/en/latest/community/contributing/testing_pull_requests.html) for more information.

Just confirmed this fixed https://github.com/godotengine/godot/issues/54343

Reply:
Also fixed https://github.com/godotengine/godot/issues/25526